e2e/wdio/core/tests/dynamic-page.e2e-spec.ts

```diff
@@ -0,0 +1,156 @@
+        const buttonsLength = getElementArrayLength(dynamicPage + button);
+        for (let i = 0; i < buttonsLength; i++) {
+            if (i !== 5 && i !== 6) {
+                expect(getElementClass(dynamicPage + button, i)).not.toContain('compact', 'size of the button still compact');
+            }
+        }
+        click(dynamicPage + button, 7);
+        for (let i = 0; i < buttonsLength; i++) {
+            if (i !== 5 && i !== 6) {
+                expect(getElementClass(dynamicPage + button, i)).toContain('compact', 'size of the button did not change to cozy');
+            }
+        }
+    });
+
+    it('should check 📍 button working', () => {
+        checkPiningArea(tabsExample);
+        checkPiningArea(tabsExample);
+        checkPiningArea(responsiveExample);
+    });
+
+    it('should check open-close flexible columns', () => {
+        click(columnLayoutExample + button);
+        click(dynamicPage + button, 7);
+        pause(1000);
+        expect(isElementDisplayed(flexileColumn, 1)).toBe(true, 'flexible column is not visible');
+        click(flexileColumn + button, 10);
+        pause(1000);
+        expect(isElementDisplayed(flexileColumn, 2)).toBe(true, 'flexible column is not visible');
+        click(flexileColumn + button, 11);
+        pause(1000);
+        expect(isElementDisplayed(flexileColumn, 0)).toBe(false, 'flexible column is not hidden');
+        expect(isElementDisplayed(flexileColumn, 1)).toBe(false, 'flexible column is not hidden');
+        expect(isElementDisplayed(flexileColumn, 2)).toBe(true, 'flexible column is not visible');
+        click(flexileColumn + button, 11);
+        pause(1000);
+        expect(isElementDisplayed(flexileColumn, 0)).toBe(true, 'flexible column is not visible');
+        expect(isElementDisplayed(flexileColumn, 1)).toBe(true, 'flexible column is not visible');
+        expect(isElementDisplayed(flexileColumn, 2)).toBe(true, 'flexible column is not visible');
+        click(flexileColumn + button, 12);
+        pause(1000);
+        expect(isElementDisplayed(flexileColumn, 2)).toBe(false, 'flexible column is not hidden');
+    });
+
+    it('should check RTL and LTR orientation', () => {
+        dynamicPagePage.checkRtlSwitch();
+    });
+
+    function checkPiningArea(section: string): void {
+        click(section + button);
+        click(pinButton);
+        expect(getAttributeByName(pinButton, 'aria-selected')).toBe('true', 'area is not pinned');
+        click(exitButton);
+    }
+
+    function checkCollapsingArea(section: string): void {
+        click(section + button);
+        click(collapseButton);
+        expect(isElementDisplayed(collapsibleHeader)).toBe(false, 'collapsible area is not hidden');
+        click(collapseButton);
+        expect(isElementDisplayed(collapsibleHeader)).toBe(true, 'collapsible area is not displayed');
+        click(exitButton);
+    }
+
+    function checkPageOpenClose(section: string): void {
```

Comment:
please make a default method here that takes input section and button
then write different it for each type
i.e.

```
it('should close page by clicking save, () => {
openPage(section); //or click(openSectionButton); or click(openPageBtn, index);
checkPageClosed(section, saveButton);
});
it('should close page by clicking close, () => {
openPage(section);
checkPageClosed(section, closeButton);
});